overlord/snapstate/snapstate_test.go

```diff
@@ -85,9 +85,26 @@ type snapmgrBaseTest struct {
 	user3 *auth.UserState
 }
 
+// state must be locked by caller
 func (s *snapmgrBaseTest) settle(c *C) {
```

Comment:
this needs to use defer s.state.Lock()   no?  otherwise it has the same problem of the original code

Reply:
Yes, a panic in the Settle would trigger the same problem. I fixed it now

Reply:
thanks